tests/basic.rs

```diff
@@ -310,3 +310,19 @@ fn schedule_counter() {
     assert_eq!(handle.tag().load(Ordering::SeqCst), 3);
     r.recv().unwrap();
 }
+
+#[test]
+fn drop_inside_schedule() {
+    let s = "1234".to_owned();
+
+    let (task, _) = async_task::spawn(
+        async {},
+        move |task| {
+            println!("{:?}", s);
+            drop(task);
+            println!("{:?}", s);
```

Comment:
How are you making sure that this test doesn't actually pass when the UAF isn't actually catched here by some random segfault or something like that? (For example on macOS that test passes easily even with the UAF-present, the output is simply wrong).

Reply:
I would suggest something like this:

```rust
use std::sync::atomic::{AtomicBool, Ordering};

fn test() {
    struct DropGuard(AtomicBool);
    impl Drop for DropGuard {
        fn drop(&mut self) {
            self.0.store(false, Ordering::SeqCst);
        }
    }
    let s = DropGuard(AtomicBool::new(true));

    let (task, _) = async_task::spawn(
        async {},
        move |task| {
            assert_eq!(s.0.load(Ordering::SeqCst), true);
            drop(task);
            assert_eq!(s.0.load(Ordering::SeqCst), true);
        },
        (),
    );
    task.schedule();
}
```

Reply:
The UAF is caught by valgrind (it is used in CI), and I checked to make sure.

That said, your version is more robust so I'll try that instead.